var/spack/repos/builtin/packages/elsi/package.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+# See the Spack documentation for more information on packaging.
+# ----------------------------------------------------------------------------
+
+from spack import *
+
+
+class Elsi(CMakePackage):
+    """ELSI provides a unified interface for electronic structure
+    codes to a variety of eigenvalue solvers."""
+
+    homepage = "https://wordpress.elsi-interchange.org/"
+    url      = "http://wordpress.elsi-interchange.org/wp-content/uploads/2019/03/elsi-2.2.1.tar.gz"
+
+    version('2.2.1', sha256='5b4b2e8fa4b3b68131fe02cc1803a884039b89a1b1138af474af66453bec0b4d')
+
+    # Variants (translation of cmake options)
+    variant(
+        'add_underscore', default=True,
+        description="Suffix C functions with an underscore"
+    )
+    variant(
+        'elpa2_kernel', default="none", description="ELPA2 Kernel",
+        values=('BGQ', 'AVX', 'AVX2', 'AVX512'), multi=False
+    )
+    variant(
+        'enable_pexsi', default=False, description='Enable PEXSI support'
+    )
+    variant(
+        'enable_sips', default=False, description='Enable SLEPc-SIPs support'
+    )
+    variant(
+        'use_external_elpa', default=False,
+        description="Build ELPA using SPACK"
+    )
+    variant(
+        'use_external_ntpoly', default=False,
+        description="Build NTPoly using SPACK"
+    )
+    variant(
+        'use_external_omm', default=False,
+        description="Use external libOMM and MatrixSwitch"
+    )
+    variant(
+        'use_external_superlu', default=False,
+        description="Use external SuperLU DIST"
+    )
+    variant(
+        'use_mpi_iallgather', default=True,
+        description="Use non-blocking collective MPI functions"
+    )
+
+    # Basic dependencies
+    depends_on('blas', type="link")
+    depends_on('lapack', type="link")
+    depends_on('cmake', type="build")
+    depends_on('mpi')
+    depends_on('scalapack', type="link")
+
+    # Library dependencies
+    depends_on('elpa', when='+use_external_elpa')
```

Comment:
@william-dawson  Reading through this again, I overlooked this logic: generally it is best to install dependencies with Spack when possible (i.e. it would be ideal to just always have `-DUSE_EXTERNAL_ELPA=ON`). This primarily becomes an issue if a package depends on `ELSI` and `ELPA` for example (in which case without `+use_external_elpa`, Spack may build two different instances of `ELPA` and you could encounter weird inconsistencies at run time.

If you anticipate that `ELSI` would be used as a dependency in Spack, I request that you default these variants to "on" (or avoid offering the option altogether).

Reply:
@scheibelp thanks for the feedback! I'll talk to the head developer of ELSI and if he agrees I'll submit a pull request with those changes.

Reply:
Oh FWIW I didn't mean to suggest changing ELSI, just that you could update the Spack package for ELSI by having it always set `-DUSE_EXTERNAL_ELPA=ON`. Without Spack, that option is probably useful, but when Spack is used, it would be ideal to always use Spack-built dependencies (where possible).

Reply:
Right, that's what I thought! I just want to get his feedback in case there are some special features for the internal version of ELPA.